### ISSUE TITLE
Sticks are incorrectly mapped in Wired Xinput in Mac OS Sequoia

### ISSUE THREAD

**Issue by zurce** [moonlight-stream/moonlight-qt#1429]
**Describe the bug**
Mac OS Sequoia introduced support for Wired XInput devices, this new support is broken as the vertical axis is inversed when using Moonlight

**Steps to reproduce**
Connect any Xinput controlled Wired

**Screenshots**
If applicable, add screenshots to help explain your problem. If the issue is related to video glitching or poor quality, please include screenshots.

**Affected games**

**Other Moonlight clients**
Mac OS Sequoia

**Additional context**
Device I'm using with this issue

Flydigi Apex 4
Mobapad M9 

They both have a DInput/Xinput switch and do work fine when switched to Dinput, but since the dongle only works in Xinput  it is an issue if wanted to use with it

**Comment by cgutman** [comment]
Please attach the logs from Moonlight after streaming with those controllers. The logs will be in your `/tmp` folder (accessible via Cmd+Shift+G in Finder, then `/tmp`).

**Comment by Danzikks** [comment]
Yes, the controller doesn't work in moonlight but in steam link it does, before sequoia it didn't work there and there.
<img width="560" alt="CleanShot 2024-10-13 at 01 22 10@2x" src="https://github.com/user-attachments/assets/ddce9e0d-1994-4ff0-b00f-2f0565cbca3b">

**Comment by Danzikks** [comment]
> Пожалуйста, прикрепите логи из Moonlight после трансляции с этими контроллерами. Логи будут в вашей `/tmp`папке (доступной через Cmd+Shift+G в Finder, затем `/tmp`).
Please enable xinput support for wired gamepads in mac os sequoia. I'm willing to help you test builds, or attach some logs.

**Comment by Danzikks** [comment]
> Пожалуйста, прикрепите логи из Moonlight после трансляции с этими контроллерами. Логи будут в вашей `/tmp`папке (доступной через Cmd+Shift+G в Finder, затем `/tmp`).

Hi! Yes, this is indeed true, the control in macos sequoia is broken via wired connection. The sticks are inverted. Please fix it, impossible to play, I'm willing to help test or send some logs
<img width="564" alt="CleanShot 2024-10-13 at 14 25 23@2x" src="https://github.com/user-attachments/assets/c03fb542-de3d-4768-b30f-47c2181c139c">

**Comment by Danzikks** [comment]
> > Пожалуйста, прикрепите логи из Moonlight после трансляции с чудовищными контроллерами. Логи будут в вашем `/tmp`файле (доступ через Cmd+Shift+G в Finder, затем `/tmp`).
> 
> Привет! Да, это действительно так, управление в macos sequoia сломано через проводное соединение. Стики перевернуты. Пожалуйста, исправьте это, невозможно играть, я готов помочь протестировать или отправить логи <img alt="CleanShot 2024-10-13 в 14 25 23@2x" width="564" src="https://private-user-images.githubusercontent.com/149537573/376030136-c03fb542-de3d-4768-b30f-47c2181c139c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg4NDkxOTQsIm5iZiI6MTcyODg0ODg5NCwicGF0aCI6Ii8xNDk1Mzc1NzMvMzc2MDMwMTM2LWMwM2ZiNTQyLWRlM2QtNDc2OC1iMzBmLTQ3YzIxODFjMTM5Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxM1QxOTQ4MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yZGZlZWQ4ZDU3ODExZjk1ZjA5ODI5NTAzMTI4YTJhZTYxNjJiZDk4NWJjOTFjMjcyOTE2OTRjMjU2Mjg3ZTQxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.LmWFyQZQpAFJrntm_zR169DDDfJHicTTJmioTYkONbQ">

To be fair, the same error occurs in steam link. The direction of stick movement is inverted along the Y axis

**Comment by alecmckeown** [comment]
https://github.com/libsdl-org/SDL/issues/11002

This seems to be a similar issue to something I experienced in Crossover for Mac OS. I found this related discussion.

**Comment by cgutman** [comment]
Yeah, that was me :)

We got it sorted on the SDL side, so the next release with updated SDL will fix it.

**Comment by LeeeSe** [comment]
Adding environment variables before running Wine can temporarily resolve this issue in Crossover or Whisky applications based on Wine.

```SDL_JOYSTICK_HIDAPI=0  wine game.exe```

**Comment by zurce** [comment]
Any update on this? It seem the reason is clear now but the issue still remain

**Comment by lzf185** [comment]
> Any update on this? It seem the reason is clear now but the issue still remain
Hello, have you found a solution to the problem? I also have the problem of Y axis inversion when playing wine games on Flydigi Dune Fox.

**Comment by Iv4nS** [comment]
Unable to play with moonlight and my controller at the moment because of this issue..

**Comment by b4iterdev** [comment]
What about adding an invert stick option in moonlight client ?

**Comment by nicolas-albert** [comment]
Same issue here 😵‍💫

**Comment by b4iterdev** [comment]
I have solved it by installing nightly build on moonlight, looks like it's fixed upstream but there haven't a release yet

**Comment by nicolas-albert** [comment]
@b4iterdev thx!

I tried with the **Moonlight-macOS-179857** from Action build (url from macos build + "xattr -cr") and it's OK!

https://github.com/moonlight-stream/moonlight-qt/commit/179857f17d55e1c1b50ab6057425c9243b6fbf48
